src/main/java/com/cloudbees/jenkins/support/SupportAction.java

```diff
@@ -146,8 +152,10 @@ public void doDownload(StaplerRequest req, StaplerResponse rsp) throws ServletEx
         if (supportPlugin != null) {
             SupportProvider supportProvider = supportPlugin.getSupportProvider();
             if (supportProvider != null) {
+                SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd_HH.mm.ss");
+                dateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
                 // let the provider name it
-                filename = supportProvider.getName();
+                filename = supportProvider.getName() + "_" + dateFormat.format(new Date());
```

Comment:
You need the date always, not just for those coming from the supportProvider. Do the date injection in the addHeader line not here